godot-core/src/engine/gfile.rs

```diff
@@ -662,33 +662,25 @@ impl Read for GFile {
             return Ok(0);
         }
 
-        let mut read_bytes = 0;
-        while read_bytes < bytes_to_read {
-            buf[read_bytes] = self.fa.get_8();
-            read_bytes += 1;
-            self.check_error()?;
-        }
+        let gd_buffer = self.fa.get_buffer(bytes_to_read as i64);
+        buf[0..gd_buffer.len()].copy_from_slice(gd_buffer.as_slice());
+
+        self.check_error()?;
 
-        Ok(read_bytes)
+        Ok(gd_buffer.len())
     }
 }
 
 impl Write for GFile {
     fn write(&mut self, buf: &[u8]) -> IoResult<usize> {
-        // TODO: Refactor to make use of `FileAccess::store_buffer()` method for higher efficiency in writing to file.
-        let bytes_to_write = buf.len();
-        let mut bytes_written = 0;
-
-        while bytes_written < bytes_to_write {
-            self.fa.store_8(buf[bytes_written]);
-            bytes_written += 1;
-            self.check_error()?;
-        }
-        Ok(bytes_written)
+        self.fa.store_buffer(PackedByteArray::from(buf));
```

Comment:
I would store this `PackedByteArray` as a field `write_buffer` in `GFile`. Keep it initially empty, and on each `write()` call, make sure it's big enough to hold `buf.len()` bytes.

This avoids re-allocating the entire buffer for every single `write()`, which should speed up writing of larger files.

Unfortunately we can't do the same for reading, since `PackedByteArray` is always _returned_ by `FileAccess::get_buffer()`. Due to copy-on-write, this means they need to reallocate it every time.

Reply:
Hmm, `FileAccess::store_buffer()` takes `PackedByteArray` not by ref, so at some point during `write()` I will need to allocate it (I think?)

Reply:
Figured it out!